sycl/test-e2e/VirtualMem/basic_access_from_kernel_virtual_mem.cpp

```diff
@@ -0,0 +1,87 @@
+// REQUIRES: aspect-ext_oneapi_virtual_mem
+
+// RUN: %{build} -o %t.out
+// RUN: %{run} %t.out
+
+#include "sycl/handler.hpp"
+#include <sycl/ext/oneapi/experimental/device_architecture.hpp>
+#include <sycl/ext/oneapi/virtual_mem/physical_mem.hpp>
+#include <sycl/ext/oneapi/virtual_mem/virtual_mem.hpp>
+
+namespace syclext = sycl::ext::oneapi::experimental;
+
+int main() {
+  sycl::queue Q;
+  sycl::context Context = Q.get_context();
+  int Failed = 0;
+
+  size_t UsedGranularityInBytes = syclext::get_mem_granularity(
+      Context, syclext::granularity_mode::recommended);
+
+  syclext::physical_mem NewPhysicalMem{Q.get_device(), Context,
+                                       UsedGranularityInBytes};
```

Comment:
Why do we only allocate "granularity" bytes? I think that we should first take some number of elements `N` of type `T` which we would like to test. Then we need to compute number of bytes required `BR = N * sizeof(T)` and maybe round it up to make sure that it is evenly divisible by "granularity" that is the value which we should be using.

I think that it will be a better test, because right now if recommended granularity returned to you is low (let's say 8 bytes), then you only test 2-element array, which doesn't seem realistic. Let's cover at least 100 elements